packages/geoadmin-layers/src/types/layers.ts

```diff
@@ -0,0 +1,428 @@
+import type { Options } from 'ol/source/WMTS'
+
+import { CoordinateSystem } from '@geoadmin/coordinates'
+import { ErrorMessage, WarningMessage } from '@geoadmin/log/Message'
+
+import type { LayerTimeConfig } from '@/types/timeConfig'
+
+export const DEFAULT_OPACITY = 1.0
+export const WMS_SUPPORTED_VERSIONS = ['1.3.0']
+
+export enum LayerType {
+    WMTS = 'WMTS',
+    WMS = 'WMS',
+    GEOJSON = 'GEOJSON',
+    AGGREGATE = 'AGGREGATE',
+    KML = 'KML',
+    GPX = 'GPX',
+    VECTOR = 'VECTOR',
+    GROUP = 'GROUP',
+    COG = 'COG',
+}
+
+export interface LayerAttribution {
+    name: string
+    url?: string
+}
+
+export interface LayerLegend {
+    url: string
+    format: string
+    width?: number
+    height?: number
+}
+
+/** @interface Layer */
+export interface Layer {
+    /** Name of this layer in the current lang */
+    name: string
+    /**
+     * The unique ID of this layer that will be used in the URL to identify it (and also in
+     * subsequent backend services for GeoAdmin layers)
+     */
+    readonly id: string
+    /** The layer type */
+    readonly type: LayerType
+    /**
+     * What's the backend base URL to use when requesting tiles/image for this layer, will be used
+     * to construct the URL of this layer later on
+     */
+    readonly baseUrl: string
+    /**
+     * Flag telling if the base URL must always have a trailing slash. It might be sometime the case
+     * that this is unwanted (i.e. for an external WMS URL already built past the point of URL
+     * params, a trailing slash would render this URL invalid). Default is `false`
+     */
+    // readonly ensureTrailingSlashInBaseUrl: boolean
+    /** Value from 0.0 to 1.0 telling with which opacity this layer should be shown on the map. */
+    opacity: number
+    /** If the layer should be visible on the map or hidden. */
+    visible: boolean /* TODO propose to rename this to isVisible */
+    /** Description of the data owner(s) for this layer. */
+    readonly attributions: LayerAttribution[]
+    /** Define if this layer shows tooltip when clicked on. */
+    readonly hasTooltip: boolean
+    /** Define if this layer has a description that can be shown to users to explain its content. */
+    hasDescription: boolean
+    /** Define if this layer has a legend that can be shown to users to explain its content. */
+    readonly hasLegend: boolean
+    /** Define if this layer comes from our backend, or is from another (external) source. */
+    readonly isExternal: boolean
+    /** Set to true if some parts of the layer (e.g. metadata) are still loading */
+    isLoading: boolean
+    /** Time series config */
+    timeConfig: LayerTimeConfig | null
```

Comment:
The time config object itself can be `readonly`, if that doesn't block its mutation in its nested attributes